File: 0167.两数之和II-输入有序数组/0167-两数之和II-输入有序数组.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int left = 0;
        int rigth = numbers.length - 1;
        while((numbers[left] + numbers[rigth]) != target)
        {
            if((numbers[left] + numbers[rigth]) < target)
            {
                left++;
            }
            else
            {
                rigth--;
            }
        }

        int res[] = new int[2];
        res[0] = left + 1;
        res[1] = rigth + 1;
        return res;
    }
}

File: 0167.两数之和II-输入有序数组/0167-两数之和II-输入有序数组.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* numbers, int numbersSize, int target, int* returnSize){
    *returnSize = 2;

    int left = 0;
    int right = numbersSize - 1;
    int *res = malloc(sizeof(int) * 2);
    res[0] = 0;
    res[1] = 0;

    while(left < right)
    {
        if((numbers[left] + numbers[right]) == target)
        {
            res[0] = left + 1;
            res[1] = right + 1;
            break;
        }
        else if((numbers[left] + numbers[right]) < target)
        {
            left++;
        }
        else
        {
            right--;
        }
    }

    return res;
}